lms/static/js/utils/dropdown.js

```diff
@@ -0,0 +1,223 @@
+
+            listenForClick: function() {
+
+                _dropdown.opts.button.on('click', function() {
+                    _dropdown.closeDropdownMenu(); // close any open menus
+                    _dropdown.openDropdownMenu($(this)); // then open the chosen menu
+                });
+
+                _dropdown.opts.page.on('click', function() {
+                    _dropdown.closeDropdownMenu();
+                });
+            },
+
+            listenForKeypress: function() {
+
+                _dropdown.opts.page.on('keydown', function(e) {
+                    var keyCode = e.keyCode,
+                        focused = $(e.currentTarget.activeElement),
+                        items, menu;
+
+                    if (27 === keyCode) {
+                        // When the ESC key is pressed, close all menus
+                        _dropdown.closeDropdownMenus();
+                    }
+
+                    if (focused.is('.action')) {
+                        // Key handlers for when a menu item has focus
+                        menu = focused.closest('.dropdown-menu');
+
+                        switch (keyCode) {
+                            case 38:
+                                _dropdown.previousMenuItemLink(focused, menu);
+                                break;
+
+                            case 40:
+                                _dropdown.nextMenuItemLink(focused, menu);
+                                break;
+
+                            case 27:
+                                _dropdown.closeDropdownMenus();
+                                break;
+                        }
+
+                    } else if (focused.is('.has-dropdown')) {
+                        // Key handlers for when the button that opens the menu has focus
+                        switch(keyCode) {
+                            case 40:
+                                _dropdown.openDropdownMenu(focused);
+                                break;
+
+                            case 13:
+                                _dropdown.openDropdownMenu(focused);
+                                break;
+
+                            case 27:
+                                _dropdown.closeDropdownMenus();
+                                break;
+                        }
+                    } else if (focused.is('.dropdown-menu')) {
+                        // Key handlers for when the menu itself has focus, before an item within it receives focus
+                        menu = focused.closest('.dropdown-menu');
+
+                        switch(keyCode) {
+                            case 40:
+                                _dropdown.focusFirstItem(menu);
+                                break;
+
+                            case 27:
+                                _dropdown.closeDropdownMenus();
+                                break;
+                        }
+                    } else {
+
+                        switch(keyCode) {
```

Comment:
I don't understand what this bottom switch is doing.

I also must admit that I don't have keyCodes memorized, so comments as to what the various numbers correlate to would be helpful.